### DISCUSSION TITLE
Key binding for Previous Window

### DISCUSSION THREAD

**Discussion by finnvyrn** [labwc/labwc#1321] [Ideas]
Currently, it's only possible to go to next window with Alt-Tab, or a defined key. Please add the feature of going to previous window.

**Comment by Narrat** [comment]
This is already possible (but maybe not with the default config).
Just add this to your `rc.xml` in the keyboard section:
``` xml
    <keybind key="A-S-Tab">
      <action name="PreviousWindow" />
    </keybind>
```

**Reply by finnvyrn** [reply]
No. This doesn't work. I have tried it by putting it to my rc.xml. You can try it yourself. Whenever you type Alt and Shift at the same time, it goes to next window.

**Reply by johanmalm** [reply]
Have you set alt+shift to toggle keyboard layout?

**Reply by finnvyrn** [reply]
No. My hotkey for switching keyboard layout is Ctrl-Shift-Space.

**Reply by Narrat** [reply]
>No. This doesn't work. I have tried it by putting it to my rc.xml. You can try it yourself. Whenever you type Alt and Shift at the same time, it goes to next window.

I don't need to try it, because I use this keybind since a few months and it is working for me. And I have no odd behaviour or window switching with only pressing Alt and Shift.
Obvious questions would be: Did you reconfigure your current running labwc session after changing the config? What does your config look like? Or at least the part regarding your key combinations.

**Reply by finnvyrn** [reply]
```
<?xml version="1.0"?>

<!--
  This is a very simple config file with many options missing. For a complete
  set of options with comments, see docs/rc.xml.all
-->

<labwc_config>

  <theme>
    <name></name>
    <keepBorder>no</keepBorder>
    <cornerRadius>8</cornerRadius>
    <font name="EXO 2" size="24" />
  </theme>

  <windowSwitcher show="yes" preview="yes" outlines="yes">
  </windowSwitcher>
  
  <keyboard>
    <default />
    <repeatRate>100</repeatRate>
    <repeatDelay>175</repeatDelay>

    <keybind key="W-c">
      <action name="Close" />
    </keybind>

    <keybind key="W-q">
      <action name="Execute" command="alacritty" />
    </keybind>

    <keybind key="S-Space">
      <action name="Execute" command="rofi -show drun" />
    </keybind>

    <keybind key="W-w">
      <action name="Execute" command="rofi -show window" />
    </keybind>

    <keybind key="F7">
      <action name="Execute" command="watershot --copy"/>
    </keybind>

    <keybind key="W-z">
      <action name="Execute" command="alacritty -e vifm"/>
    </keybind>

    <keybind key="XF86_MonBrightnessUp">
      <action name="Execute" command="brightnessctl set +1%" />
    </keybind>

    <keybind key="XF86_MonBrightnessDown">
      <action name="Execute" command="brightnessctl set 1%-" />
    </keybind>
  </keyboard>

  <mouse>
    <context name="Frame">
      <mousebind button="A-Left" action="Press">
      </mousebind>
      <mousebind button="A-Left" action="Drag">
      </mousebind>
      <mousebind button="A-Right" action="Press">
      </mousebind>
      <mousebind button="A-Right" action="Drag">
      </mousebind>
    </context>
  </mouse>

  <windowRules>
    <windowRule title="*"><action name="Maximize"/></windowRule>
  </windowRules>

</labwc_config>
```

**Reply by finnvyrn** [reply]
I don't include 

```
    <keybind key="A-S-Tab">
      <action name="PreviousWindow" />
    </keybind>
```

in the rc.xml I just sent here, but I did include it and try, and it didn't work.

**Reply by Narrat** [reply]
Really odd then. I used your config (just replacing the apps in some of the keybinds) in a nested session and `NextWindow` and `PreviousWindow` worked like I expected.
The only thing I have is an issue with the osd from the tab-switcher (which throws `00:01:20.016 [../labwc-0.7.0/src/desktop.c:458] Unknown node detected` and stays sometimes visible).
Something else that could be special on your side?

**Reply by johanmalm** [reply]
@finnvyrn Works here too with the `PreviousWindow` action as above. If you run in VM-ware that might consume the keybind I guess... otherwise don't know what to suggest.

> The only thing I have is an issue with the osd from the tab-switcher (which throws `00:01:20.016 [../labwc-0.7.0/src/desktop.c:458] Unknown node detected` and stays sometimes visible).

Ah, yeah, that's just when the pointer is over the OSD window-switcher. We ought to sort that out - probably by giving the window-switcher a node-descriptor.

**Comment by Gnomasz** [comment]
> I don't include
> 
> ```
>     <keybind key="A-S-Tab">
>       <action name="PreviousWindow" />
>     </keybind>
> ```
> 
> in the rc.xml I just sent here, but I did include it and try, and it didn't work.

Are there any other key bindings that you didn't include in the file you sent? Maybe something is interfering?

Or maybe your keys are somehow interpreted improperly by the system. Have you tried checking how your "Alt" and "Shift" show in `wev`?
